x-pack/legacy/plugins/siem/public/components/open_timeline/open_timeline.tsx

```diff
@@ -31,56 +44,145 @@ export const OpenTimeline = React.memo<OpenTimelineProps>(
+
+    const { getBatchItemsPopoverContent } = useEditTimelinBatchActions({
+      deleteTimelines,
+      dispatchToaster,
+      selectedItems,
+      tableRef,
+    });
+
+    const nTimelines = useMemo(
+      () => (
+        <FormattedMessage
+          id="xpack.siem.open.timeline.showingNTimelinesLabel"
+          defaultMessage="{totalSearchResultsCount} {totalSearchResultsCount, plural, one {timeline} other {timelines}} {with}"
+          values={{
+            totalSearchResultsCount,
+            with: (
+              <span data-test-subj="selectable-query-text">
+                {query.trim().length ? `${i18n.WITH} "${query.trim()}"` : ''}
+              </span>
+            ),
+          }}
+        />
+      ),
+      [totalSearchResultsCount, query]
+    );
+
+    return (
+      <>
+        <EditTimelineActions
+          actionItem={actionItem}
+          deleteTimelines={deleteTimelines}
+          exportedIds={exportedIds}
+          getExportedData={exportSelectedTimeline}
+          isDeleteTimelineModalOpen={isDeleteTimelineModalOpen}
+          isEnableDownloader={isEnableDownloader}
+          onComplete={onCompleteEditTimelineAction}
+        />
+
+        <EuiPanel className={OPEN_TIMELINE_CLASS_NAME}>
+          <TitleRow
+            data-test-subj="title-row"
+            onAddTimelinesToFavorites={onAddTimelinesToFavorites}
+            selectedTimelinesCount={selectedItems.length}
+            title={title}
+          >
+            <SearchRow
+              data-test-subj="search-row"
+              onlyFavorites={onlyFavorites}
+              onQueryChange={onQueryChange}
+              onToggleOnlyFavorites={onToggleOnlyFavorites}
+              query={query}
+              totalSearchResultsCount={totalSearchResultsCount}
+            />
+          </TitleRow>
+
+          <UtilityBar border>
+            <UtilityBarSection>
+              <UtilityBarGroup>
+                <UtilityBarText data-test-subj="query-message">
+                  <>
+                    {i18n.SHOWING} {nTimelines}
+                  </>
+                </UtilityBarText>
+              </UtilityBarGroup>
 
-      <SearchRow
-        data-test-subj="search-row"
-        onlyFavorites={onlyFavorites}
-        onQueryChange={onQueryChange}
-        onToggleOnlyFavorites={onToggleOnlyFavorites}
-        query={query}
-        totalSearchResultsCount={totalSearchResultsCount}
-      />
+              <UtilityBarGroup>
+                <UtilityBarText>{i18n.SELECTED_TIMELINES(selectedItems.length)}</UtilityBarText>
+                <UtilityBarAction
+                  iconSide="right"
+                  iconType="arrowDown"
+                  popoverContent={getBatchItemsPopoverContent}
+                >
+                  {i18n.BATCH_ACTIONS}
+                </UtilityBarAction>
+                <UtilityBarAction
+                  iconSide="right"
+                  iconType="refresh"
+                  onClick={useCallback(() => {
```

Comment:
I never saw that before but I am pretty sure that it needs to move up in a const handler